### PR TITLE
prisma-fmt: Accept multiple files as an input for native types hints

### PR DESCRIPTION
**Pull request by SevInf**
Autocomplete for native types is broken at the moment if schema folder
is used.

### CODE REVIEW

**Review by Jolg42 — APPROVED** [review]
Looks good, minus the dbg statement

File: prisma-fmt/src/native.rs [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
     };
 
+    let schema: Vec<_> = schema.into();
+    let validated_configuration = match psl::parse_configuration_multi_file(&schema) {
+        Ok((_, validated_configuration)) => validated_configuration,
+        Err(_) => return "[]".to_owned()
+    };
+
     if validated_configuration.datasources.len() != 1 {
+        dbg!(3);
```
[the remark this answers is not in the document]
Comment (SevInf) [reply prisma-fmt/src/native.rs:18]: nope

File: prisma-fmt/tests/native_types.rs [hunk 2]
```diff
@@ -9,9 +9,31 @@ fn test_native_types_list_on_crdb() {
...
...A","_number_of_args":0,"_number_of_optional_args":0,"prisma_types":["Bytes"]},{"name":"Timestamp","_number_of_args":0,"_number_of_optional_args":1,"prisma_types":["DateTime"]},{"name":"Timestamptz","_number_of_args":0,"_number_of_optional_args":1,"prisma_types":["DateTime"]},{"name":"Date","_number_of_args":0,"_number_of_optional_args":0,"prisma_types":["DateTime"]},{"name":"Time","_number_of_args":0,"_number_of_optional_args":1,"prisma_types":["DateTime"]},{"name":"Timetz","_number_of_args":0,"_number_of_optional_args":1,"prisma_types":["DateTime"]},{"name":"Boolean","_number_of_args":0,"_number_of_optional_args":0,"prisma_types":["Boolean"]},{"name":"Bit","_number_of_args":0,"_number_of_optional_args":1,"prisma_types":["String"]},{"name":"VarBit","_number_of_args":0,"_number_of_optional_args":1,"prisma_types":["String"]},{"name":"Uuid","_number_of_args":0,"_number_of_optional_args":0,"prisma_types":["String"]},{"name":"Xml","_number_of_args":0,"_number_of_optional_args":0,"prisma_types":["String"]},{"name":"Json","_number_of_args":0,"_number_of_optional_args":0,"prisma_types":["Json"]},{"name":"JsonB","_number_of_args":0,"_number_of_optional_args":0,"prisma_types":["Json"]}]"#]];
```
Comment (Druue) [inline prisma-fmt/tests/native_types.rs:37]: I'm also wondering, is this supposed to just be the whole set of native types? or native types for the string field? Would it maybe be better to separate the expects based on prisma type, and then the expects are also easier to read

**Comment by Jolg42** [comment]
@SevInf one clippy error for you https://github.com/prisma/prisma-engines/actions/runs/9320644281/job/25657868351?pr=4898#step:4:55
```
error: useless conversion to the same type: `std::vec::Vec<(std::string::String, psl::SourceFile)>`
Error:   --> prisma-fmt/src/native.rs:11:26
   |
11 |     let schema: Vec<_> = schema.into();
   |                          ^^^^^^^^^^^^^ help: consider removing `.into()`: `schema`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
   = note: `-D clippy::useless-conversion` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::useless_conversion)]`

```

**Comment by Jolg42** [comment]
and one rustfmt error
https://github.com/prisma/prisma-engines/actions/runs/9320644281/job/25657870271?pr=4898#step:4:64

File: prisma-fmt/tests/native_types.rs (hunk 2 above)
Comment (SevInf) [reply prisma-fmt/tests/native_types.rs:37]: I think as of right now, LT gets whole list of native types and filters them out themselves.
Ideally, of course, rust would just handle that completion, but i think parser is not quite there to support it,

**Pull request merged**